packages/steps/src/BuildStepInput.ts

```diff
@@ -110,21 +110,16 @@ export class BuildStepInput<
     }
 
     const valueDoesNotRequireInterpolation =
-      rawValue === undefined ||
-      typeof rawValue === 'boolean' ||
-      typeof rawValue === 'number' ||
-      typeof rawValue === 'object';
+      rawValue === undefined || typeof rawValue === 'boolean' || typeof rawValue === 'number';
     if (valueDoesNotRequireInterpolation) {
-      const currentTypeName =
-        typeof rawValue === 'object' ? BuildStepInputValueTypeName.JSON : typeof rawValue;
-      if (currentTypeName !== this.allowedValueTypeName && rawValue !== undefined) {
+      if (typeof rawValue !== this.allowedValueTypeName && rawValue !== undefined) {
         throw new BuildStepRuntimeError(
           `Input parameter "${this.id}" for step "${this.stepDisplayName}" must be of type "${this.allowedValueTypeName}".`
         );
       }
       return rawValue as BuildStepInputValueTypeWithRequired<T, R>;
     } else {
-      const valueInterpolatedWithGlobalContext = this.ctx.interpolate(rawValue);
+      const valueInterpolatedWithGlobalContext = this.ctx.interpolate(rawValue as string | object);
```

Comment:
I think it's not, it's probably a leftover from some previous troubleshooting

Reply:
Actually no, it's because of an error `Argument type undefined is not assignable to parameter type string | object`. Which I'm not sure why is happening since we already check if `rawValue` is defined

Reply:
Interesting 🤔 

If we do `assert(rawValue !== undefined)` does the error still occur?

Reply:
That works